### DISCUSSION TITLE
Reference to ACGBPoly

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#26783] [Q&A Modules: Phase field]
Hello, 
Can you please provide some references for a kernel 'ACGBPoly' the way it is formulated here on MOOSE?
Thank you

**Comment by GiudGiud** [comment]
should be from the references listed here
https://mooseframework.inl.gov/modules/phase_field/Grain_Growth_Model.html

**Reply by GiudGiud** [reply]
actually for grain boundaries might be those
https://mooseframework.inl.gov/modules/phase_field/Grain_Boundary_Anisotropy.html

**Reply by ashishdhole** [reply]
Thank you Guillaume,
I understand the kernel is based on particle pinning model explained by Moelans et al. 2005 (https://doi.org/10.1016/j.actamat.2004.12.026), But in the kernel, the equation is modified a bit by adding a few terms, which was not explained in the paper mentioned above. I just want to have a little more insight on how the kernel is derived. If you can please provide the papers it would be highly appreciated.

**Reply by GiudGiud** [reply]
I would not know. Maybe @laagesen

**Reply by laagesen** [reply]
Hi @ashishdhole, the best reference to understand where that kernel comes from is 

https://iopscience.iop.org/article/10.1088/0965-0393/23/4/045009

It may have been based on the Moelans paper you mentioned and it is cited but I think Moelans paper uses a slightly different from that Tonks et al used. The `ACGBPoly` kernel implements the variational derivative of the last term in Equation (4) of that paper, which is mu* a_s * sum_i (c^2 * eta_i^2) (the mu comes from equation 3). For a particular Order parameter, say eta_1, the variational derivative of that term wrt eta_1 is 2 * mu * a_s * c^2 * eta_1. So that will be computed in the function `computeDFDOP` in the switch statement with case `Residual`. You will see `c * c` there for c^2 and `_u[_qp]` for eta_1 (the member variable `_u` represents the nonlinear variable for this kernel, which should be eta_1). The prefactor `mult` is computed as 2 * mu * a_s at the beginning of that function. The notation `gamma` is a bit confusing here because in the paper, it is used for interfacial energy, whereas in the code `gamma` is the prefactor for the eta_I^2 * eta_j^2 term (called a_GB in the paper).

**Reply by ashishdhole** [reply]
Thank you @laagesen  for the clarification and the reference.
